### PR TITLE
CAL-234: Default security attributes get set on metacards if they contain no highwater security markings, regardless of policy

### PR DESCRIPTION
**Pull request by Lambeaux**
#### What does this PR do?
Updates the `DefaultSecurityAttributeValuesPlugin` to no longer skip setting defaults when metacards have a non-empty policy. This change was needed due to the introduction of the `ResourceUriPolicy` plugin. 

#### Who is reviewing it?
@shaundmorris 
@brendan-hofmann 
@ryeats 

#### Choose 2 committers to review/merge the PR.
@kcwire 
@stustison

#### How should this be tested?
Ingest a variety of test data, of different formats, and ensure that everything receives security attributes where appropriate. The attributes should render in the UI when inspecting a metacard. 

#### What are the relevant tickets?
[CAL-234](https://codice.atlassian.net/browse/CAL-234)

#### Checklist:
- [ ] Documentation Updated
	- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java [hunk 1]
```diff
@@ -108,24 +108,29 @@ Subject getSystemSubject() {
...
-     * Adds default system high-water markings in the event that none of the policies were able to apply security markings to this metacard.
+     * Adds default system high-water markings in the event that none of the policies were able to
+     * apply security markings to this metacard.
      *
      * @return Map of the metacard security markings to the value of their corresponding system high
      * attribute.
      */
     public Metacard addDefaults(Metacard metacard) {
 
+        Map<String, Attribute> securityMarkings = getHighwaterSecurityMarkings();
```
Comment (ryeats) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java]: I think we can cache this and just update it anytime the configuration changes

File: catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 2]
```diff
@@ -15,21 +15,28 @@
...
+                <entry key="codewords" value="FineAccessControls"/>
+                <entry key="disseminationControls" value="disseminationControls"/>
+                <entry key="otherDisseminationControls" value="otherDisseminationControls"/>
+                <entry key="ownerProducer" value="ownerProducer"/>
+            </map>
+        </argument>
     </bean>
 
-    <service ranking="99" ref="defaultMarkingsPlugin" interface="ddf.catalog.plugin.PreIngestPlugin"/>
+    <service ranking="99" ref="defaultMarkingsPlugin"
```
Comment (bdeining) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Is there are reason to have this `ranking=99`?

Comment (Lambeaux) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: That's not something I added. It was already there. @ryeats can you enlighten us?

Comment (ryeats) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: I think originally I wanted it to run after anything that could update the metacards security markings. I'm not sure that still holds water because you would think those would be run in earlier plugins.

Comment (bdeining) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: I think we should be cautious using the service ranking; this might lead to confusion later on.    There's no guarantee of ordering for the PreIngestPlugins, so if this needs to be ranked last, we should add a comment as to why.

Comment (Lambeaux) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: @bdeining since the service ranking has not been changed in this PR, do we want to mess with it? Seems like an unrelated change with minimal added value that could bite us after release.

Comment (bdeining) [inline catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: I would just add a comment here that the ranking is here so the plugin occurs last during processing (if that still is the case).  Otherwise, this might bite someone down the road when they develop a PreIngestPlugin that they want firing last (and can't figure out why it's not).

**Comment by brjeter** [comment]
👍 Successfully heroed!
